### DISCUSSION TITLE
Livewire add to cart issue

### DISCUSSION THREAD

**Discussion by rolinbos** [duncanmcclean/simple-commerce#1114] [Help]
Hi,

I'm using Livewire and want to add item to cart. But I get the message "CSRF token mismatch".

This is the code for making the HTTP request:

```
        $response = Http::withHeaders([
            'Accept' => 'application/json',
            'Content-Type' => 'application/json',
            'X-CSRF-TOKEN' => csrf_token(),
        ])->post(config('app.url') . '/!/simple-commerce/cart-items', [
            '_token' => csrf_token(),
            '_request' => encrypt('Empty'),
            '_error_redirect' => encrypt('Empty'),
            '_redirect' => encrypt('Empty'),
            'product' => $this->id,
            'quantity' => 1,
            'variant' => 'variant-one',
        ]);
```

The csrf_token() and the {{ csrf_token() }} (in my blade) are the same. Does someone have any idea how I can fix this?
Or is there maybe another option to call a method inside PHP to update the card?

**Comment by duncanmcclean** [comment]
Even though you're providing the CSRF token in the HTTP request, there's probably something else session-related you're missing in the request which is causing it not to work.

Instead of making an HTTP request, it might be easier to add the line item manually, using [Simple Commerce's PHP APIs](http://simple-commerce.duncanmcclean.com/php-api#content-orders):

```php
use DuncanMcClean\SimpleCommerce\Orders\Cart\Drivers\CartDriver;

class YourComponent extends Component
{
    use CartDriver;

    public function doSomething()
    {
        $this->getCart()->addLineItem([
            'product' => 'product-id',
            'variant' => 'variant-id',
            'quantity' => 1,
            'total' => 1500,
        ])->save();
    }
}
```

_^ I haven't tested this code snippet, but I believe it should work._
